packages/contracts/contracts/voting/multisig/Multisig.sol

```diff
@@ -0,0 +1,426 @@
+        bool _tryExecution
+    ) external returns (uint256 proposalId) {
+        uint64 snapshotBlock = getBlockNumber64() - 1;
+
+        if (!isListedAtBlock(_msgSender(), snapshotBlock)) {
+            revert ProposalCreationForbidden(_msgSender());
+        }
+
+        proposalId = proposalCounter.current();
+
+        // Create the proposal
+        Proposal storage proposal_ = proposals[proposalId];
+
+        proposal_.open = true;
+        proposal_.parameters.snapshotBlock = snapshotBlock;
+        proposal_.parameters.minApprovals = minApprovals_;
+        proposal_.tally.addresslistLength = addresslistLengthAtBlock(snapshotBlock); // TODO https://aragonassociation.atlassian.net/browse/APP-1417
+
+        for (uint256 i; i < _actions.length; ) {
+            proposal_.actions.push(_actions[i]);
+            unchecked {
+                ++i;
+            }
+        }
+
+        _incrementProposalCount();
+
+        if (_approveProposal) {
+            approve(proposalId, _tryExecution);
+        }
+
+        emit ProposalCreated({
+            proposalId: proposalId,
+            creator: _msgSender(),
+            metadata: _metadata,
+            actions: _actions
+        });
+    }
+
+    /// @notice Approves and, optionally, executes the proposal.
+    /// @param _proposalId The ID of the proposal.
+    /// @param _tryExecution If `true`, execution is tried after the approval cast. The call does not revert if execution is not possible.
+    function approve(uint256 _proposalId, bool _tryExecution) public {
+        address approver = _msgSender();
+        if (!_canApprove(_proposalId, approver)) {
+            revert ApprovalCastForbidden(_proposalId, approver);
+        }
+
+        Proposal storage proposal_ = proposals[_proposalId];
+
+        unchecked {
+            proposal_.tally.approvals += 1;
+        }
+
+        proposal_.approvers[approver] = true;
+
+        emit Approved({proposalId: _proposalId, approver: approver});
+
+        if (_tryExecution && _canExecute(_proposalId)) {
+            _execute(_proposalId);
+        }
+    }
+
+    /// @notice Checks if an account can participate on a proposal vote. This can be because the vote
+    /// - was executed, or
+    /// - the voter is not listed.
+    /// @param _proposalId The proposal Id.
+    /// @param _account The address of the user to check.
+    /// @return bool Returns true if the account is allowed to vote.
+    ///@dev The function assumes the queried proposal exists.
+    function canApprove(uint256 _proposalId, address _account) external view returns (bool) {
+        return _canApprove(_proposalId, _account);
+    }
+
+    /// @notice Checks if a proposal can be executed.
+    /// @param _proposalId The ID of the proposal to be checked.
+    /// @return True if the proposal can be executed, false otherwise.
+    function canExecute(uint256 _proposalId) external view returns (bool) {
+        return _canExecute(_proposalId);
+    }
```

Comment:
Why don't we simply define the function itself and avoid one dummy function jump? It can be resolved if it is because of how interfaces are defined